### PR TITLE
Feat/integration parameters resolution

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
Waiting on https://github.com/superfaceai/one-sdk-js/pull/166 

## Description

This PR refactors integration parameters resolution during bind

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/client/profile-provider.ts [hunk 1]
```diff
@@ -339,13 +341,61 @@ export class ProfileProvider {
...
 
+  private resolveIntegrationParameters(
+    superJsonParameters: { [key: string]: string } | undefined,
+    providerJsonParameters: IntegrationParameter[]
+  ): Record<string, string> | undefined {
+    if (superJsonParameters === undefined) {
+      return undefined;
+    }
+
+    //TODO: super. json defines parameters but provider.json does not - error?
```
[the remark this answers is not in the document]
Comment (lukas-valenta) [reply src/client/profile-provider.ts:360]: also delete the comment

File: src/internal/parser.ts [hunk 2]
```diff
@@ -76,44 +76,50 @@ export class Parser {
...
-      `${info.profileName}-[0-9a-f]+${EXTENSIONS.profile.build}`,
-      cachePath
-    );
+    await Parser.clearFileCache(path);
 
     // And write parsed file to cache
-    const parsedProfile = parseProfile(new Source(input, fileName));
+    parsedProfile = parseProfile(new Source(input, fileName));
+    if (!isProfileDocumentNode(parsedProfile)) {
+      //TODO: more helpful error - can this be product of not matching AST package versions?
```
Comment (lukas-valenta) [inline src/internal/parser.ts:115]: yes, do

File: src/internal/parser.ts [hunk 3]
```diff
@@ -123,36 +129,40 @@ export class Parser {
...
+    const loaded = JSON.parse(
+      await fsp.readFile(path, { encoding: 'utf8' })
+    ) as unknown;
+    //Check if valid type
+    if (!guard(loaded)) {
+      return undefined;
+    }
+    //Check if checksum match
+    if (loaded.astMetadata.sourceChecksum !== sourceHash) {
+      console.log('CHECK');
```
Comment (lukas-valenta) [inline src/internal/parser.ts:154]: log

**Review by lukas-valenta — CHANGES_REQUESTED** [review]

File: src/internal/parser.ts (hunk 2 above)
Comment (Jakub-Vacek) [reply src/internal/parser.ts:115]: FIxed in: https://github.com/superfaceai/one-sdk-js/pull/166/commits/120f1b8f355ae6e288b8e2f3aaecdd8b1ca2bb2c

File: src/internal/parser.ts (hunk 3 above)
Comment (Jakub-Vacek) [reply src/internal/parser.ts:154]: FIxed in: https://github.com/superfaceai/one-sdk-js/pull/166/commits/120f1b8f355ae6e288b8e2f3aaecdd8b1ca2bb2c

**Review by lukas-valenta — APPROVED** [review]

**Pull request merged**